.github/workflows/reftest.yml

```diff
@@ -0,0 +1,64 @@
+name: reftest
+
+on: [push, pull_request]
+
+env:
+  cache-version: v1
+
+jobs:
+  reftest:
+    runs-on: ${{ matrix.os }}
+    name: Node ${{ matrix.node }} / ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node: [12.x, 14.x]
+    steps:
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Use npm@7
+        run: npm i -g npm@7 --registry=https://registry.npmjs.org
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ env.cache-version }}-${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ env.cache-version }}-${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ env.cache-version }}-${{ runner.os }}-build-
+            ${{ env.cache-version }}-${{ runner.os }}-
+      - name: Checkout akashic-engine repository
+        uses: actions/checkout@v2
+        with:
+          path: akashic-engine
+      - name: Checkout engine-files repository
+        uses: actions/checkout@v2
+        with:
+          repository: akashic-games/engine-files
+          path: engine-files
+      - name: Pack akashic-engine
+        working-directory: akashic-engine
+        id: akashic_engine
+        run: |
+          npm ci --ignore-scripts
+          echo "::set-output name=pack_name::$(npm pack)"
+      - name: Run engine-files reftest
+        working-directory: engine-files
+        run: |
+          npm ci
+          npm i ../akashic-engine/${{steps.akashic_engine.outputs.pack_name}} --no-save
+          npm run build:full:parts
+          npm run build:canvas:parts
+          npm test
+      - name: Archive artifact
+        if: ${{ always() }}
```

Comment:
workflow の知識が乏しいのでドキュメント見ながら確認したのですが、自信がないので確認させてください。
ここの `always()` (成果部をアップロードするステップ)は、これより前のステップの checkout で致命的なエラー(チェックアウトできなかった等)が起きた場合は実行されない認識であってますでしょうか？

Reply:
いえ、実行されるかと思います。どちらかというと前 step の `npm test` で致命的なエラー (exit 0 以外を返したなど) が発生した際に内容を確認したいために指定しており、そこにエラー内容による区別はされないはずです。

仮に checkout 部分でエラーが発生した場合は path に指定したファイルが存在しないことになりますが、その際この step はデフォルトで warning だけを出して正常終了するようです。

https://github.com/actions/upload-artifact#customization-if-no-files-are-found
> If a path (or paths), result in no files being found for the artifact, the action will succeed but print out a warning. In certain scenarios it may be desirable to fail the action or suppress the warning. The if-no-files-found option allows you to customize the behavior of the action if no files are found.

Reply:
>仮に checkout 部分でエラーが発生した場合は path に指定したファイルが存在しないことになりますが、その際この step はデフォルトで warning だけを出して正常終了するようです。
なるほど。ありがとうございます。